templates/core/terraform/api-webapp/api-webapp.tf

```diff
@@ -0,0 +1,215 @@
+  app_settings = {
+
+    "APPINSIGHTS_INSTRUMENTATIONKEY" = azurerm_application_insights.core.instrumentation_key
+  }
+
+  site_config {
+    app_command_line            = "gunicorn -w 2 -k uvicorn.workers.UvicornWorker main:app"
+    remote_debugging_enabled    = false
+    scm_use_main_ip_restriction = true
+    cors {
+      allowed_origins     = []
+      support_credentials = false
+    }
+    always_on       = true
+    min_tls_version = "1.2"
+    ip_restriction {
+      action     = "Deny"
+      ip_address = "0.0.0.0/0"
+      name       = "Deny all"
+      priority   = 2147483647
+    }
+    websockets_enabled = false
+  }
+
+  logs {
+    application_logs {
+      file_system_level = "Information"
+    }
+
+    http_logs {
+      file_system {
+        retention_in_days = 7
+        retention_in_mb   = 100
+      }
+    }
+  }
+}
+
+resource "azurerm_user_assigned_identity" "management_api" {
+  name                = "id-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+}
+
+resource "azurerm_private_endpoint" "management_api_private_endpoint" {
+  name                = "pe-api-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  subnet_id           = var.shared_subnet
+  private_service_connection {
+    private_connection_resource_id = azurerm_app_service.management_api.id
+    name                           = "psc-api-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+    subresource_names              = ["sites"]
+    is_manual_connection           = false
+  }
+  private_dns_zone_group {
+    name                 = "privatelink.azurewebsites.net"
+    private_dns_zone_ids = [azurerm_private_dns_zone.azurewebsites.id]
+  }
+}
+
+resource "azurerm_private_dns_zone" "azurewebsites" {
+  name                = "privatelink.azurewebsites.net"
+  resource_group_name = var.resource_group_name
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "azurewebsites" {
+  resource_group_name   = var.resource_group_name
+  virtual_network_id    = var.core_vnet
+  private_dns_zone_name = azurerm_private_dns_zone.azurewebsites.name
+  name                  = "azurewebsites-link"
+  registration_enabled  = false
+}
+
+resource "azurerm_app_service_virtual_network_swift_connection" "api-integrated-vnet" {
+  app_service_id = azurerm_app_service.management_api.id
+  subnet_id      = var.web_app_subnet
+}
+
+resource "azurerm_monitor_diagnostic_setting" "webapp_management_api" {
+  name                       = "diag-${var.resource_name_prefix}-${var.environment}-${var.tre_id}"
+  target_resource_id         = azurerm_app_service.management_api.id
+  log_analytics_workspace_id = var.log_analytics_workspace_id
+
+  log {
+    category = "AppServiceHTTPLogs"
+    enabled  = true
+
+
+    retention_policy {
+      days    = 1
```

Comment:
Logging policies probably need further consideration. Another issue maybe?

Reply:
Yeah, we should have a story on what logging policies in general are. @christoferlof would you mind adding a story and then resolving this with it?

Reply:
Please see #50 .